### PR TITLE
Improve contribution documentation

### PR DESCRIPTION
**Pull request by omri374**
Rearranged stuff, moved the grpc stuff to the genproto repo (see PR there). Emphasized what's optional and what's not

### CODE REVIEW

**Comment by omri374** [comment]
Please also review https://github.com/microsoft/presidio-genproto/pull/29

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+   When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Use custom logic
+
+   In some cases, rule-based logic provides the best way of detecting entities. The Presidio EntityRecognizer API allows you to use spaCy extracted features like lemmas, part of speech, dependencies and more to create your logic. When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Deep learning based methods
+
+   Deep learning methods offer excellent detection rates for NER. They are however more complex to train, deploy and tend to be slower than traditional approaches. When contributing a DL based method, the best option would be to create a sidecar container which is isolated from the presidio-analyzer container. On the `presidio-analyzer` side, one would extend the [RemoteRecognizer](presidio-analyzer/analyzer/remote_recognizer.py) class and implement the network interface between `presidio-analyzer` and the endpoint of the model's container.
+
```
Comment (eladiw) [inline CONTRIBUTING.md]: Great explanation. Also add that any contribution to the supported types should be added here:
https://github.com/microsoft/presidio/blob/master/docs/field_types.md

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Presidio's strength lies in its ability to be completely pluggable and customizable. Adding new types of databases, stream-analytics engines and data stores is extremely useful for many users.
+While we're working on adding more documentation on this type of customization, you can refer to these documents and resources:
+1. Scheduler
+
+   - The [scheduler documentation](docs/tutorial_scheduler.md) describes how to configure the data source and target for a periodic job
+
+2. Templates for source, target and stream configurations and service definitions, on the [presidio-genproto](https://github.com/microsoft/presidio-genproto) repo. For example: 
+
+	- [data sink protocols](https://github.com/microsoft/presidio-genproto/blob/master/src/datasink.proto). Specifically, the [DatasinkTypesEnum](https://github.com/microsoft/presidio-genproto/blob/1734e2635c253f79e4c44398315d92fe9d084601/src/datasink.proto#L37)
```
Comment (eladiw) [inline CONTRIBUTING.md]: any reason this is not master?

File: docs/custom_fields.md [hunk 3]
```diff
@@ -2,134 +2,138 @@
...
 
-    ```sh
-    POST <api-service-address>/api/v1/analyzer/recognizers/<new_recognizer_name>
-    ```
+  c. Creating a new recoginzer
 
-    ### Request structure:
+  ```sh
+  POST <api-service-address>/api/v1/analyzer/recognizers/<new_recognizer_name>
+  ```
```
Comment (eladiw) [inline docs/custom_fields.md]: d) updating a recognizer is exactly the same as c) only its with PUT

File: CONTRIBUTING.md (hunk 1 above)
Comment (omri374) [reply CONTRIBUTING.md]: Added

File: CONTRIBUTING.md (hunk 2 above)
Comment (omri374) [reply CONTRIBUTING.md]: It's a permalink, probably making this robust to changes in the target file. I can copy the original link with reference to the row but I guess that's the preferred way.

File: docs/custom_fields.md (hunk 3 above)
Comment (omri374) [reply docs/custom_fields.md]: Added

**Pull request merged**